devicetypes/smartthings/aeotec-doorbell-6.src/aeotec-doorbell-6.groovy

```diff
@@ -0,0 +1,231 @@
+ */
+metadata {
+	definition(name: "Aeotec Doorbell 6", namespace: "SmartThings", author: "SmartThings", mnmn: "SmartThings") {
+		capability "Actuator"
+		capability "Health Check"
+		capability "Tamper Alert"
+		capability "Alarm"
+
+		fingerprint mfr: "0371", prod: "0003", model: "00A2", deviceJoinName: "Aeotec Doorbell 1"
+	}
+
+	tiles {
+		multiAttributeTile(name: "alarm", type: "generic", width: 6, height: 4) {
+			tileAttribute("device.alarm", key: "PRIMARY_CONTROL") {
+				attributeState "off", label: 'off', action: 'alarm.siren', icon: "st.alarm.alarm.alarm", backgroundColor: "#ffffff"
+				attributeState "both", label: 'ring!', action: 'alarm.off', icon: "st.alarm.alarm.alarm", backgroundColor: "#0e7507"
+			}
+		}
+		standardTile("off", "device.alarm", inactiveLabel: false, decoration: "flat") {
+			state "default", label: '', action: "alarm.off", icon: "st.secondary.off"
+		}
+		valueTile("tamper", "device.tamper", height: 2, width: 2, decoration: "flat") {
+			state "clear", label: 'tamper clear', backgroundColor: "#ffffff"
+			state "detected", label: 'tampered', backgroundColor: "#ffffff"
+		}
+
+		main "alarm"
+		details(["alarm", "off", "tamper"])
+	}
+}
+
+private getNumberOfSounds() {
+	def numberOfSounds = [
+			"0003" : 8 //Aeotec Doorbel 6
+	]
+	return numberOfSounds[zwaveInfo.prod] ?: 1
+}
+
+def installed() {
+	initialize()
+	sendEvent(name: "alarm", value: "off", isStateChange: true, displayed: false)
+}
+
+def updated() {
+	initialize()
+}
+
+def initialize() {
+	if (!childDevices) {
+		addChildren(numberOfSounds)
+	}
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	}
+	log.debug "Parse returned: ${result.inspect()}"
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	createEvent(name: "alarm", value: cmd.value ? "both" : "off")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	def endpoint = cmd.sourceEndPoint
+	if (endpoint == 1) {
+		zwaveEvent(encapsulatedCommand)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def securedEncapsulatedCommand = cmd.securedEncapsulatedCommand()
+	if (securedEncapsulatedCommand) {
+		zwaveEvent(securedEncapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+
+}
+private secure(physicalgraph.zwave.Command cmd) {
+	zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+}
+
+private secureEncap(cmd, endpoint = null) {
+	secure(encap(cmd, endpoint))
+}
+
+private onOffCmd(value, endpoint = null) {
+			secureEncap(zwave.basicV1.basicSet(value: value), endpoint)
```

Comment:
nit: spacing

Reply:
between onOffCmd and bracket? In other methods I don't have space between name and bracket

Reply:
No. 3 tabs before `secureEncap`